data-serving/scripts/converters.py

```diff
@@ -0,0 +1,121 @@
+'''Library of conversion functions for converting CSV line-list data to json.'''
+
+import datetime
+import logging
+import math
+import numbers
+import pandas as pd
+from typing import Any, Callable, Dict, List
+
+
+VALID_SEXES = ['female', 'male', 'other']
+
+range_ct = 0
+single_ct = 1
+
+
+def convert_range(value: str, parse_fn: Callable[[str], str], format_fn: Callable[[Any], Any]):
+    '''
+    Converts either a string representation of a single value or a range of values into a range object. The values can be of any types using the custom parse and format functions.
+    '''
+    if pd.isna(value):
+        return None
+
+    # First check if the value is represented as a range.
+    if type(value) is str and value.find('-') >= 0:
```

Comment:
Are any other delimiters used (e.g. `:`, `..`)?

Reply:
not that i've seen yet, for these fields. I think : is used in other fields i haven't mapped yet, for lists (like symptoms)

i've gotten surprisingly few parser errors on the whole dataset -- a few invalid dates, a few ages with strings in them ("9 months")

Reply:
Interesting, thanks!